src/Microsoft.Data/DataFrame.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrame 
+    /// </summary>
+    public partial class DataFrame
+    {
+        internal DataFrameTable _table;
+        public DataFrame()
+        {
+            _table = new DataFrameTable();
+        }
+
+        public long RowCount => _table.RowCount;
+        public int ColumnCount => _table.ColumnCount;
+        public IList<string> Columns()
```

Comment:
Although general OOP principle is to return the base type (or interface) for future flexibility, I am not sure we generally do that in API design. There is some perf impact (although probably immeasurable in most cases). Anyway it doesn't matter - easy to adjust during API review - but if you're interested, ask Immo